addons/udes_stock/models/stock_move.py

```diff
@@ -310,10 +310,20 @@ def _action_assign(self):
         don't return any extra moves that may have been created
         by refactoring.
         """
+        Move = self.env["stock.move"]
+
         res = super(StockMove, self)._action_assign()
 
         assign_moves = self.exists()._action_refactor(stage="assign")
 
+        # Unreserve any partially reserved lines if not allowed by the picking type
```

Comment:
Is this the same code used in `_reserve_stock_assign()`? can we abstract it in a function? Have we considered if the code in the other function is still needed, since it will call this function anyway.